### PR TITLE
VB-208: Fix/placeholder

### PR DESCRIPTION
**Pull request by double-hope**

### CODE REVIEW

File: packages/frontend/components/verification/component.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+              {variant == 2 ? (
+                <SuccessModal></SuccessModal>
+              ) : (
+                <>{closeModal()}</>
+              )}
+            </>
+          )}
+        </>
+      )}
+    </Modal>
```
Comment (Nightlngale) [inline packages/frontend/components/verification/component.tsx:41]: switch case or { key1: <Component1 />, ... }[key]

File: packages/frontend/components/verification/enter-code/component.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          variant="primary"
+          type="text"
+          name="phone"
+        />
+        <Button onClick={changeModal}>
+          {t('verify:enterCode.button.continue')}
+        </Button>
+      </div>
+
+      <InternalLink
```
Comment (Nightlngale) [inline packages/frontend/components/verification/enter-code/component.tsx:50]: Semantically not correct. LinkButton

File: packages/frontend/components/verification/enter-code/component.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  const previousModal = (e) => {
+    e.preventDefault();
+    dispatch(previousVerifyModal());
+  };
+
+  const changeModal = (e) => {
+    e.preventDefault();
+    dispatch(nextVerifyModal());
+    console.log(variant);
```
Comment (Nightlngale) [inline packages/frontend/components/verification/enter-code/component.tsx:23]: logs

File: packages/frontend/pages/_app.tsx [hunk 4]
```diff
@@ -18,10 +20,15 @@ type AppPropsWithLayout = AppProps & {
...
 
+  const { isModal } = useTypedSelector((state) => state.verify);
+  const { hasToken } = useAuth();
+  const isMounted = useMounted();
+
   return (
     <ThemeProvider>
       <UserProvider>{getLayout(<Component {...pageProps} />)}</UserProvider>
       <PagesLoader />
+      {isMounted && hasToken && isModal && <VerificationModal />}
```
Comment (Nightlngale) [inline packages/frontend/pages/_app.tsx:31]: Please move all of these checks inside your component

File: packages/frontend/public/locales/en/index.ts [hunk 5]
```diff
@@ -19,4 +20,5 @@ export const en = {
   auth,
   item,
   rules,
+  verify,
```
Comment (Nightlngale) [inline packages/frontend/public/locales/en/index.ts:23]: can be a part of common

File: packages/frontend/store/verify/actions.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { VerifyActionTypes } from './action-types';
+
+export interface VerifyAction {
+  type: VerifyActionTypes;
+  payload?: any;
```
Comment (Nightlngale) [inline packages/frontend/store/verify/actions.ts:5]: why any

**Review by Nightlngale — CHANGES_REQUESTED** [review]

File: packages/frontend/components/verification/component.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+  useEffect(() => {
+    setIsVisible(true);
+  }, []);
+
+  const closeModal = () => {
+    dispatch(hideVerifyModal());
+  };
+
+  const renderSwitch = (param) => {
```
Comment (Nightlngale) [inline packages/frontend/components/verification/component.tsx:24]: renderContent

File: packages/frontend/components/verification/component.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    switch (param) {
+      case 0:
+        return <EnterPhoneModal />;
```
Comment (Nightlngale) [inline packages/frontend/components/verification/component.tsx:27]: It's not a modal - it's modal's content

File: packages/frontend/components/verification/styles.ts [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+  & span {
+    font-family: inherit;
+    font-size: ${fontSizes.body2};
+    font-weight: ${fontWeights.body2};
+    line-height: ${lineHeights.body2};
+    color: ${colors.textLight};
+  }
+`;
+export const headline = ({
```
Comment (Nightlngale) [inline packages/frontend/components/verification/styles.ts:37]: add empty line

File: packages/frontend/components/verification/success-verification/component.tsx [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+export default function SuccessModal() {
+  const dispatch = useAppDispatch();
+  const { variant } = useTypedSelector((state) => state.verify);
+  const { t } = useTranslation();
+
+  const changeModal = (e) => {
+    e.preventDefault();
+    dispatch(nextVerifyModal());
+    console.log(variant);
```
Comment (Nightlngale) [inline packages/frontend/components/verification/success-verification/component.tsx:17]: log

**Review by Nightlngale — APPROVED** [review]

File: packages/backend/src/services/verify.ts [hunk 11]
```diff
@@ -91,6 +91,7 @@ export class VerifyService {
     const user = await this._userRepository.getById(userId);
     await this.deleteCodeByType(userId, type);
     const code = await this.createVerificationCode(userId, type);
+    console.log(`phone: ${code}`);
```
Comment (Nightlngale) [inline packages/backend/src/services/verify.ts:94]: log

File: packages/frontend/components/auth/verification/phone-verification.tsx [hunk 12]
```diff
@@ -1,22 +1,24 @@
 import { Button } from '@primitives';
 import { useTranslation } from 'next-i18next';
-import { useAppDispatch, useAuth, useTypedSelector } from '@hooks';
+import {
+  useAppDispatch,
+  // useAuth,
```
Comment (Nightlngale) [inline packages/frontend/components/auth/verification/phone-verification.tsx:5]: remove comments

File: packages/frontend/components/auth/verification/phone-verification.tsx [hunk 13]
```diff
@@ -41,20 +43,20 @@ export const PhoneVerification = () => {
...
+        {/*<div css={verifyText}>*/}
+        {/*  <span>*/}
+        {/*    {t('auth:phoneText')}*/}
+        {/*    {user && (*/}
+        {/*      <span css={verifyEntity}> {hideMainTextPart(user.phone)}</span>*/}
+        {/*    )}*/}
+        {/*    !*/}
+        {/*  </span>*/}
+        {/*  <span>{t('auth:enterCode')}!</span>*/}
+        {/*</div>*/}
```
Comment (Nightlngale) [inline packages/frontend/components/auth/verification/phone-verification.tsx:55]: remove comments

File: packages/frontend/tsconfig.json [hunk 14]
```diff
@@ -12,7 +12,7 @@
       "@hooks": ["hooks"],
       "@hocs": ["hocs"],
       "@helpers": ["helpers"],
-      "@components/*": ["components/*"],
+      "@components": ["components/*"],
```
Comment (Nightlngale) [inline packages/frontend/tsconfig.json:15]: why?

**Review by Nightlngale — CHANGES_REQUESTED** [review]

File: packages/frontend/components/auth/verification/phone-verification.tsx (hunk 13 above)
Comment (dmitriy-uvin) [reply packages/frontend/components/auth/verification/phone-verification.tsx:55]: it must be used

File: packages/frontend/components/auth/verification/phone-verification.tsx (hunk 12 above)
Comment (dmitriy-uvin) [reply packages/frontend/components/auth/verification/phone-verification.tsx:5]: also must be used, that's why i commented it

**Pull request merged**